BUILD.md

```diff
@@ -0,0 +1,42 @@
+## TL;DR
+```
+docker build -t aleph-node/build -f docker/Dockerfile_build . && \
+docker run -ti --volume=$(pwd):/node/build aleph-node/build
```

Comment:
Doesn't work for me without sudo.

Reply:
Add your user to docker group and relog: https://docs.docker.com/engine/install/linux-postinstall/#manage-docker-as-a-non-root-user

Reply:
The point is that this is a guide for users, and it should be completely idiot-proof. If that's a necessary step, then it should be part of the guide.

Reply:
also without `docker` ;)

Reply:
I am not sure with this. On one hand I don't want to encourage people to use docker via `sudo` and without `user-namaspaces`, on the other I don't how much detailed should be the dependencies described here.

Reply:
Also `/node/nix-build.sh: line 28: patchelf: command not found`

Reply:
Well, the issue is you can use docker as you want. In rootless mode, with your user added to the group or as a root. If we want to be clear, we can link to the doc that I commented above and let user decide.

Reply:
I don't want to make it to become `a guide on how to install docker`, so perhaps something like: requirements - being able to run `docker` locally (I don't care whether you are using root account).

Reply:
Ideally the TL-DR should also specify where I can find the binary afterwards.

Reply:
`patchelf` thing should be fixed now - sorry for `force` push. I just realized last moment that it shouldn't be part of the `nix.shell` script, which should just describe how to build it on nix (that patchelf there was making it broken).

Reply:
I can tell you with high certainty that what a user wants here is a list of commands to run and the location of the binary afterwards. That's it. They just want it to work.

An explanation what the commands do and stuff is a nice bonus, but most users will ignore this part unless something is not working properly. Familiarity with docker might vary.

Reply:
Sounds cool, but also little naive. There is also nothing mentioned about which shell we are using throughout this guide, and for sure it want work under `fish` due to how we call `pwd`. Any idea what should be some sane requirements. I can just add `sudo su` at the start, but it won't look good.

Reply:
Just take a look at other repos of comparable projects, like solana, near, polkadot to see how they do it. They have lots of users and their guides already got feedback and were refined a few times, so they are worth inspecting.

Reply:
I did :) https://github.com/paritytech/polkadot#build-from-source-with-docker

Reply:
`TL;DR` should now have a comment about where binary is stored afterwards.

Reply:
Actually, solana's and polkadot's build guides are really bad and totally not idion-proof.

Reply:
OK, so I will not insist to add `sudo` here, because as you say it's stupid, but mark me: normally when building they give you at least 2 alternatives: either you build with `cargo`, or with `docker`. If I don't like docker, I go ahead with cargo or whatever other alternative. Here I'm forced to use docker, and the commands, as given don't work out of the box :)

Reply:
After building
`damian@damian-HP:~/(...)/aleph-node$ ./aleph-node --help`
`./aleph-node: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by ./aleph-node)`

Reply:
I would expect that `BUILD.md` comprehensively explains how to build, and not fall back to the building section of `README.md`, unless I misunderstood the point of this guide.

Reply:
`We also prepared a configuration for the `nix` package manager that automates this process. For details please consult the [BUILD][BUILD] guide.` So, it's not a comprehensive, all types of preferences and systems, type of guide.

Reply:
Ok, fine. I think I misunderstood what this PR is supposed to do, sorry for that. I thought it solves `AZ-524`, but now I see that not quite.  You can ignore my comments -- they would make sense for a general build guide, not a guide for the nix build process.

Reply:
I am aware of this problem with `glibc` - I will try to figure it out somehow. Since it seems it is our only runtime dependency, we should make it as requirement (can be also the nix way), try to statically build with it, or allow user to link with its local glibc.

Reply:
I think your interpretation of `AZ-524` is just different than my. That's why I also made that document on notion describing it, to see what are others opinions. It's not wrong or anything and I will try to enhance this document.

Reply:
What I'm saying is that this dockerized nix build method is great, and I think many people will use it, but there will also be a high percentage of users who will want to just run `cargo build` without docker. So ideally the build guide should have two sections: 1) using cargo, 2) using nix/docker. And while 2) is done, I think 1) requires some work, because atm the instructions from 1) link to a not quite up-to-date external guide. We should ideally include complete steps (installing rust and dependencies) in our readme (or build guide, whatever works) for 1).

Reply:
Please see the new version of BUILD.md

Reply:
That looks pretty good, but still the `manual way` lacks concrete steps. I'm finding it hard to understand why would you choose to write a longish explanation why things make not work instead of just giving concrete steps that work 90 out of 100 times and 100 out of 100 times on a popular linux distribution like ubuntu. What we need here, is 1) how to install rust/cargo and git, 2) something like "Compiling and Running a Node without Docker" here https://docs.near.org/docs/develop/node/validator/running-a-node .

Reply:
I think there are good reasons for putting the disclaimers there – we don't really want to support this, even if itusually works. Although I do agree on adding clearer steps.

Perhaps leave the longish explanation in place and add clearer steps after?

Reply:
Yeah, the explanation itself makes sense, but not if the steps are not there :)

Reply:
Please see the new version of BUILD.md